src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs

```diff
@@ -145,9 +145,11 @@ public void Dispose()
             }
         }
 
+#pragma warning disable CA1821 // Remove empty Finalizers
```

Comment:
See my comment there, finalizers with only asserts should have #if !DEBUG around them. Only Contract.Requires should have #if !CONTRACTS_FULL. The analysis is correctly identifying a perf issue when these calls get compiled away.